railties/test/application/rake_test.rb

```diff
@@ -141,8 +141,40 @@ def test_rake_routes_with_controller_environment
         end
       RUBY
 
-      ENV['CONTROLLER'] = 'cart'
-      output = Dir.chdir(app_path){ `bin/rails routes` }
+      output = Dir.chdir(app_path){ `CONTROLLER=cart bin/rake routes CONTROLLER=cart` }
```

Comment:
Man, you really want to find that `CONTROLLER` don't you? :trollface:

I think one is enough :smile: